projects/driver_controls/src/event_arbiter.c

```diff
@@ -0,0 +1,57 @@
+#include <stdio.h>
+
+#include "event_arbiter.h"
+#include "input_event.h"
+#include "log.h"
+
+static FSM *s_driver_fsms[EVENT_ARBITER_MAX_FSMS];
+static EventArbiterCheck s_event_checks[EVENT_ARBITER_MAX_FSMS];
+
+static uint8_t s_num_active_fsms = 0;
+
+static bool prv_event_permitted(Event *e) {
+  bool permitted;
+
+  for (uint8_t i = 0; i < s_num_active_fsms; i++) {
+    if (s_event_checks[i] == NULL) {
+      permitted = true;
```

Comment:
If you're always setting permitted to true here, there probably isn't a point in splitting the permitted check into another conditional.

Something like
````
if (s_event_checks[i] != NULL && !s_event_checks[i](e)) {
  return false;
}
````
would probably work.